scanner/identifier.go

```diff
@@ -0,0 +1,102 @@
+package scanner
+
+import (
+	"fmt"
+
+	"github.com/grafana/river/token"
+)
+
+// IsValidIdentifier returns true if the given string is a valid river
+// identifier.
+func IsValidIdentifier(in string) bool {
+	s := New(nil, []byte(in), nil, 0)
+	_, tok, lit := s.Scan()
+	return tok == token.IDENT && lit == in
+}
+
+type SanitizeIdentifierOptions struct {
+	// EmptyValue is what to use as the identifier when the input is empty.
+	// This must be a valid river identifier.
+	EmptyValue string
+
+	// Prefix is what will be prepended to the identifier if it does not start
+	// with a letter or underscore. This must be a valid river identifier.
+	Prefix string
+
+	// Replacement is what will be used to replace invalid characters. This
+	// must be a valid river identifier or empty.
+	Replacement string
+}
```

Comment:
I think the sanitize logic here is doing a bit too much. I'm not sure it ever makes sense to have a replacement other than `_`, and I think the EmptyValue bit is separate from sanitization.

Reply:
I can return an error on empty input if we don't want it configurable.

I think it is kind of weird to force replace everything with any specific char the more I thought about it. The most obvious potential is just stripping characters by providing an empty "" Replacement:
"label!" => "label" rather than "label_"

More of a stretch, but for the purpose of discussion...
I may want "label%" => "label_percent" with knowledge about my input data format where I can anticipate each of my inputs end with "%" and doesn't have any other replacements needed.

I tried to handle all this via the defaults if you pass nil for opts but WDYT?

Reply:
that all being said, we can make more things configurable in the future as a non breaking change if we maintain the SanitizeIdentifierOptions type and add to it later.